prow/plugins/cherrypickapproved/cherrypickapproved.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cherrypickapproved
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/labels"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// PluginName defines this plugin's registered name.
+	PluginName = labels.CpApproved
```

Comment:
I think it might be better if we hard-code this to "cherry-pick-approved", we do the same in the [cherry-pick-unapproved plugin](https://github.com/kubernetes/test-infra/blob/63b8f838e6fcb32cc81b3d753cbc763f6ea336ae/prow/plugins/cherrypickunapproved/cherrypick-unapproved.go#L39). If the label name eventually gets changed in the future, it might break Prow which is an unintended side-effect

Reply:
Took this inspiration from the LGTM plugin, but I can change the const to text-only.

https://github.com/kubernetes/test-infra/blob/36a7c91fffc75a3bc9dd8d18a6a6471e7b3a9527/prow/plugins/lgtm/lgtm.go#L38